app/Libraries/Blog.php

```diff
@@ -0,0 +1,280 @@
+<?php namespace App\Libraries;
+
+use App\Entities\Post;
+use App\Exceptions\BlogException;
+use CodeIgniter\Exceptions\PageNotFoundException;
+use Config\Blog as BlogConfig;
+use League\CommonMark\CommonMarkConverter;
+
+/**
+ * Class Blog
+ *
+ * Handles reading and parsing content posts.
+ * Posts are markdown files versioned with the
+ * site repo, with a simple yaml header section
+ * to define meta data.
+ */
+class Blog
+{
+    /**
+     * @var BlogConfig
+     */
+    protected $config;
+
+    public function __construct()
+    {
+        $this->config = config(BlogConfig::class);
+    }
+
+    /**
+     * Gets the posts, sorted by most recent first.
+     * If $category is present, will locate within a
+     * subfolder of that name.
+     *
+     * @param int         $limit
+     * @param int         $offset
+     * @param string|null $category
+     *
+     * @throws BlogException
+     */
+    public function getRecentPosts(int $limit=5, int $offset=0, string $category = null)
```

Comment:
It feels a littler weird to have a config setting for `perPage` and then use a hard-coded default here in `int $limit=5`. At the very least I would like these defaults to match (10 in config, 5 here). Preferably this could accept `null` and then use the already-available config setting.

Reply:
Also safe to specify return type: `... null): array`